webpack.config.babel.js

```diff
@@ -62,38 +63,43 @@ const baseStyleLoaders = (modules=true) => [
         autoprefixer(),
         url({ url: "inline" })
       ],
-      sourceMap: true
+      sourceMap: withSourceMap
     }
   },
   {
     loader: 'less-loader',
     options: {
-      sourceMap: true
+      sourceMap: withSourceMap
     }
   }
 ];
 
 
 
-const baseStyleRules = (disableExtractToFile = false) =>
- [{
-   rule: 'exclude',
-   modules: true
- },
- {
-   rule: 'include',
-   modules: false
- }].map(({rule, modules})=> ({
-   test: /\.(less|css)$/,
-   [rule]: [`${__dirname}/node_modules`],
-   use:
-    [
-      disableExtractToFile || !PRODUCTION_BUILD ?
-        'style-loader' :
-        MiniCssExtractPlugin.loader,
-      ...baseStyleLoaders(modules)
-    ]
- }))
+const baseStyleRules = (options={}) => {
+  const { disableExtractToFile, withSourceMap } = options
+  return (
+    [{
+      rule: 'exclude',
+      modules: true
+    },
+    {
+      rule: 'include',
+      modules: false
+    }].map(({rule, modules})=> ({
+      test: /\.(less|css)$/,
+      [rule]: [`${__dirname}/node_modules`],
+      use:
+       [
+         disableExtractToFile || !PRODUCTION_BUILD ?
+           'style-loader' :
+           MiniCssExtractPlugin.loader,
+         ...baseStyleLoaders(modules, withSourceMap)
```

Comment:
this will disable the sourcemap for both dist and npm

Reply:
> From reading [webpack-contrib/css-loader#652](https://github.com/webpack-contrib/css-loader/issues/652) it seems they fixed, but according to your test it didn't correct?

Yes, I saw the source code with the fix but the problem was still there on upgrading. I couldn't figure out if I need to pass a parameter to the previous loader to make it work.
 
> Also, found this [webpack-contrib/less-loader#80 (comment)](https://github.com/webpack-contrib/less-loader/issues/80#issuecomment-287880842)
> 
> > Starting with less-loader 4, source maps contain absolute paths now.
> 
> what?? why?? 😨

Yes, I read that but from the most recent discussions it seems like they've changed their minds
https://github.com/webpack-contrib/css-loader/issues/886#issuecomment-456851963

Reply:
> this will disable the sourcemap for both dist and npm
Oops, will have a look

Reply:
I agree with @rfreitas on looking at the full file `baseStyleRules` is used in both the `configDist` and `configNpmLib`

Reply:
Yes, I had a default value in place then I did some refactoring and I removed it by mistake

Reply:
it makes sense to me now 👍